File: pyg-manager-interface/src/main/java/com/pyg/manager/service/BrandService.java
package com.pyg.manager.service;

import com.pyg.pojo.TbBrand;
import com.pyg.utils.PageResult;
import com.pyg.utils.PygResult;

import java.util.List;
import java.util.Map;

/**
 * Created by Abyss on 2018/6/12.
 * description:
 */
public interface BrandService {

    /**
     * 需求：查询所有品牌数据
     */
     List<TbBrand> findAll();

    /**
     * 需求：分页查询品牌列表
     *
     * @param page
     * @param rows
     * @return PageResult
     */
     PageResult findPage(Integer page, Integer rows);

    /**
     * 需求：添加品牌数据
     */
     PygResult add(TbBrand brand);

    /**
     * 需求：修改品牌
     */
     PygResult updateByPrimaryKey(TbBrand brand);

    /**
     * 需求：根据id查询品牌数据，用于修改回显
     */
     TbBrand findOne(Long id);

    /**
     * 需求：根据id删除品牌数据
     */
     PygResult delete(Long[] ids);

    List<Map> findBrandList();
}

File: pyg-shop-web/src/main/webapp/js/service/brandService.js
//定义service服务层
app.service("brandService", function($http) {
	// 查询所有
	this.findAll = function() {
		return $http.get("../brand/findAll");
	};

	// 分页查询
	this.findPage = function(page, rows) {
		return $http.get("../brand/findPage?page=" + page + "&rows=" + rows);
	};
	// 添加
	this.add = function(entity) {
		return $http.post("../brand/add", entity);
	}
	// 修改
	this.update = function(entity) {
		return $http.post("../brand/update", entity);
	};

	// 根据id查询品牌数据
	this.findOne = function(id) {
		return $http.get("../brand/findOne?id=" + id);
	};

	// 删除
	this.dele = function(ids) {
		return $http.get("../brand/delete?ids=" + ids);
	}
	//查询品牌数据，加载多项选择下拉框
	this.findBrandList = function(){
		return $http.get("../brand/findBrandList");
	}

});